extensions/cl_khr_unified_svm.asciidoc

```diff
@@ -0,0 +1,1069 @@
+
+        The default value is `0`, which indicates no special access behavior for
+        the host or the device for this allocation.
+
+| `CL_SVM_ALLOC_ALIGNMENT_KHR`
+  | `size_t`
+      | Specifies the minimum alignment in bytes for the SVM allocation.
+        The alignment must be a power of two and must be equal to or smaller
+        than the size of the largest data type supported by any OpenCL device in
+        _context_.
+
+        The default value is `0`, which specifies an alignment that is equal to
+        the size of the largest data type supported by any OpenCL device in
+        _context_.
+
+|====
+
+===== Freeing SVM Allocations
+
+The function
+
+[source]
+----
+cl_int  clSVMFreeWithPropertiesKHR(
+            cl_context context,
+            const cl_svm_free_properties_khr* properties,
+            cl_svm_free_flags_khr flags,
+            void* ptr);
+----
+
+frees an SVM allocation with optional properties.
+
+_context_ is a valid OpenCL context used to free the SVM allocation.
+
+_properties_ is an optional list of allocation properties and their corresponding values.
+The list is terminated with the special property `0`.
+If no free properties are required, _properties_ may be `NULL`.
+This extension does not define any free properties.
+
+_flags_ is used to specify how the SVM allocation is freed.
+This extension does not define any free flags.
+
+_ptr_ is the SVM allocation to free.
+It must be a value returned by *clSVMAlloc*, *clSVMAllocWithPropertiesKHR*, or a `NULL` pointer.
+It is the responsibility of the application to make sure enqueued commands that use _ptr_ are complete before freeing _ptr_.
+Behavior is undefined if a previously enqueued command that may be using _ptr_ is still executing.
+If _ptr_ is `NULL` then no action occurs.
+
+*clSVMFreeWithPropertiesKHR* will return `CL_SUCCESS` if the function executes successfully.
+Otherwise, it returns one of the following errors:
+
+* `CL_INVALID_CONTEXT` if _context_ is not a valid context.
+* `CL_INVALID_PROPERTY` if a memory property name in _properties_ is not a supported property name, if the value specified for a supported property name is not valid, or if the same property name is specified more than once.
+* `CL_INVALID_VALUE` if _flags_ contains an invalid SVM free flag.
+* `CL_INVALID_VALUE` if _ptr_ is not a value returned by *clSVMAlloc*, *clSVMAllocWithPropertiesKHR*, or a `NULL` pointer.
+* `CL_OUT_OF_RESOURCES` if there is a failure to allocate resources required by the OpenCL implementation on the device.
+* `CL_OUT_OF_HOST_MEMORY` if there is a failure to allocate resources required by the OpenCL implementation on the host.
+
+[NOTE]
+====
+Whether *clSVMFree* or *clSVMFreeWithPropertiesKHR* is blocking or non-blocking is unspecified.
+Applications should not rely on *clSVMFree* or *clSVMFreeWithPropertiesKHR* for synchronization, nor assume that *clSVMFree* or *clVMFreeWithPropertiesKHR* cannot cause deadlocks.
```

Comment:
As discussed in the 2025/03/18 memory subgroup TSG, I think this note should be reworded to make it clear that
- Deadlocks can only happen when **clSVMFree** (or **clSVMFreeWithPropertiesKHR**) is called on an allocation that may be used by commands pending execution.
- "blocking" means "waiting for pending commands that may use the allocation to complete execution"

With these changes, this would be allowed by the "behavior is undefined" statement above but, strictly speaking, the current wording for the note suggests that deadlocks could happen in any scenarios.